File: algorithms/interview-preparation-kit-2/php/code-response/src/FunnyString/FunnyString.php
<?php

namespace FunnyString;

class FunnyString
{
  public static function check($s)
  {
    $codes = [];
    foreach (\str_split($s) as $char) {
      $codes[] = \ord($char);
    }
    $reverse = \array_reverse($codes);
    for ($i = 0; $i < strlen($s) - 1; $i++) {
      if (\abs($codes[$i + 1] - $codes[$i]) !== \abs($reverse[$i + 1] - $reverse[$i]))
        return 'Not Funny';
    }
    return 'Funny';
  }
}

File: algorithms/interview-preparation-kit-2/ruby/code-response/src/FunnyString/funny_string.rb
class FunnyString
  def FunnyString.check(s) 
    reverse = s.split('').map {|l| l.ord}
    string = s.split('').map{|l| l.ord}
    reverse.reverse!

    for i in 0...(string.length-1) do
      if ((string[i + 1] - string[i]).abs != (reverse[i + 1] - reverse[i]).abs) 
        return 'Not Funny'
      end
    end
    return 'Funny'
  end
end